app/javascript/react/ansibleCatalog/ansiblePlaybookFields.jsx

```diff
@@ -0,0 +1,344 @@
+  }
+  removeExtraVar(index) {
+    const currentVars = [...this.state.extraVarsEditingState].splice(index,1);
+    this.setState({extraVarsEditingState: currentVars});
+  }
+  render() {
+    const { prefix, data, changeField, addExtraVars, children, formValues } = this.props;
+    const { isProvisioning, extraVarsEditingState } = this.state;
+    const dropdowns = data.dropdowns || {};
+    const repoField = `${prefix}_repository_id`;
+    const extraVarsField = `${prefix}_extra_vars`;
+    const formFields = [
+      {
+        name: `${prefix}_playbook_id`,
+        label: __('Playbook'),
+        component: FinalFormSelect,
+        options: dropdowns[`${prefix}_playbooks`] || [],
+        placeholder: DEFAULT_PLACEHOLDER,
+        validateOnMount: isProvisioning,
+        condition: { when:repoField, not: ''},
+        validate: isProvisioning ? required({ msg: __('Required') }) : undefined
+      },
+      {
+        name: `${prefix}_credential_id`,
+        label: __('Machine Credential'),
+        component: FinalFormSelect,
+        options: dropdowns.machineCredentials || [],
+        placeholder: DEFAULT_PLACEHOLDER,
+        validateOnMount: isProvisioning,
+        validate: isProvisioning ? required({ msg: __('Required') }) : undefined
+      },
+      {
+        name: `${prefix}_vault_credential_id`,
+        label: __('Vault Credential'),
+        component: FinalFormSelect,
+        options: dropdowns.vaultCredentials || [],
+        placeholder: DEFAULT_PLACEHOLDER
+      },
+      {
+        name: `${prefix}_hosts`,
+        label: __('Hostnames'),
+        component: FinalFormTextArea
+      },
+      {
+        name: `${prefix}_execution_ttl`,
+        label: __('Max TTL (mins)'),
+        component: FinalFormField
+      },
+      {
+        name: `${prefix}_log_output`,
+        label: __('Logging Output'),
+        component: FinalFormSelect,
+        options: LOGGING_OUTPUT_OPTIONS,
+        placeholder: DEFAULT_PLACEHOLDER
+      },
+      {
+        name: `${prefix}_become_enabled`,
+        component: FinalFormSwitch,
+        label: __('Escalate Privileges'),
+        onText: __('Yes'),
+        offText: __('No'),
+      },
+      {
+        name: `${prefix}_verbosity`,
+        label: __('Verbosity'),
+        component: FinalFormSelect,
+        options: VERBOSITY_OPTIONS,
+        placeholder: DEFAULT_PLACEHOLDER
+      }
+    ];
+    return (
+      <div className="row">
+        <div className="col-md-12 col-lg-6">
+          <Field
+            label={__('Repository')}
+            name={`${prefix}_repository_id`}
+            validateOnMount={isProvisioning}
+            validate={isProvisioning ? required({ msg: __('Required') }) : undefined}
+            render={({ input, meta }) =>
+              <FinalFormSelect
+                inputColumnSize={8}
+                labelColumnSize={3}
+                name={`${prefix}_repository_id`}
+                searchable={true}
```

Comment:
You don't need to explicitly set boolean props. Replace `searchable={true}` with just `searchable`